File: src/main/webapp/resources/core/js/services/tipo.embalagem.service.js
app.factory('TipoEmbalagemService', ['$http', function($http){
	
	var _findAll = function(){
		return $http.get(SERVER_APP + '/tipoembalagem/findall');
	}
	
	var _save = function(categoria){
		return $http.post(SERVER_APP + '/tipoembalagem/', categoria);
	}
	
	var _delete = function(id){
		return $http.get(SERVER_APP + '/tipoembalagem/delete/' + id);
	}
	
	return {
		
		findAll: _findAll,
		
		save: _save,
		
		delete: _delete
		
	}
}])

File: src/main/java/com/stockmanager/services/TipoEmbalagemService.java
package com.stockmanager.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.stockmanager.entities.TipoEmbalagem;
import com.stockmanager.repositories.TipoEmbalagemRepository;

@Service
public class TipoEmbalagemService {

	@Autowired
	private TipoEmbalagemRepository tipoEmbalagemRepository;
	
	public List<TipoEmbalagem> findAll(){
		return (List<TipoEmbalagem>) this.tipoEmbalagemRepository.findAll();
	}
	
	public TipoEmbalagem save(TipoEmbalagem tipoEmbalagem){
		TipoEmbalagem saved = this.tipoEmbalagemRepository.save(tipoEmbalagem);
		return saved;
	}
	
	public void delete(TipoEmbalagem tipoEmbalagem){
		this.tipoEmbalagemRepository.delete(tipoEmbalagem);
	}
}
